README.adoc

```diff
@@ -0,0 +1,232 @@
+== OpenShift S2I Builder for Vert.x
+
+This Source-to-Image Builder let's you create projects using Vert.x 3 and built with:
+
+* maven
+* gradle
+
+This builder expect the build process to generate a (unique) "fat" jar (with a file ending with `-fat.jar`).
+
+NOTE: If a project has a pom.xml and a build.gradle, maven will take precedence
+
+== ENV Options
+
+* *BUILDER_ARGS*: Allows you to specify options to pass to maven or gradle
+* *OUTPUT_DIR*: The directory where the "fat-jar" is located. It can be adjusted
+when building a multi-module projects.
+
+== Defaults
+If you do not specify any BUILDER_ARGS, by default the s2i image will use the following:
+
+* Maven
+
+----
+BUILDER_ARGS="package -Popenshift -DskipTests"
+OUTPUT_DIR=target
+----
+
+* Gradle
+
+----
+BUILDER_ARGS="build -x test"
+OUTPUT_DIR=build/libs
+----
+
+== Test in OpenShift
+
+* First load all the needed resources in a project.
+
+----
+oc create -f https://raw.githubusercontent.com/vert-x3/vertx-openshift-s2i/initial-work/vertx-s2i-all.json
```

Comment:
This will break after merge :) it links to this branch, once merged the link will be broken